aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/AwsReachability.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine;
+
+import com.amplifyframework.core.async.Cancelable;
+import com.amplifyframework.core.reachability.Host;
+import com.amplifyframework.core.reachability.Reachability;
+import com.amplifyframework.core.reachability.SocketHost;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposables;
+
+/**
+ * A utility to wait for AWS to become available.
+ */
+final class AwsReachability {
+    private static final String AWS_WEBSITE_HOST_NAME = "aws.amazon.com";
```

Comment:
Is this the right endpoint for testing reachability? `aws.amazon.com` is our brochure site, and I'm surprised we'd be using that. I'd expect this to be a ping endpoint of the AWS AppSync endpoint to specifically test our ability to hit that. Other nightmare scenarios include: our wild success and reachability tests impairing the Operational Excellence of that web application.

Reply:
https://github.com/aws-amplify/aws-sdk-android/commit/7e04c0269e59f8eb1db9c98b59a0197b63501aba

It's not COMPLETELY unprecedented I guess, but I did have the same concern as you when I first made this commit. There hasn't been any issue with it so far though.

Reply:
@raphkim Hmm. Kind of a different situation. Here we're using an endpoint for testing in the library from customer applications, not just in a test running somewhere.